File: apps/patter-voice-sdk/libraries/typescript/src/tts/inworld.ts
/** Inworld TTS for Patter pipeline mode. */
import {
  InworldTTS as _InworldTTS,
  type InworldAudioEncoding,
  type InworldDeliveryMode,
  type InworldModel,
} from "../providers/inworld-tts";

/** Constructor options for the Inworld `TTS` adapter. */
export interface InworldTTSOptions {
  /** Inworld Base64 auth token. Falls back to INWORLD_API_KEY env var. */
  apiKey?: string;
  model?: InworldModel | string;
  voice?: string;
  language?: string;
  audioEncoding?: InworldAudioEncoding | string;
  sampleRate?: number;
  bitrate?: number;
  temperature?: number;
  speakingRate?: number;
  deliveryMode?: InworldDeliveryMode | string;
  baseUrl?: string;
}

/**
 * Inworld TTS — defaults to the TTS-2 model.
 *
 * @example
 * ```ts
 * import * as inworld from "getpatter/tts/inworld";
 * const tts = new inworld.TTS();                        // reads INWORLD_API_KEY
 * const tts = new inworld.TTS({ apiKey: "...", voice: "Olivia", language: "en" });
 * ```
 */
export class TTS extends _InworldTTS {
  static readonly providerKey = "inworld";
  constructor(opts: InworldTTSOptions = {}) {
    const key = opts.apiKey ?? process.env.INWORLD_API_KEY;
    if (!key) {
      throw new Error(
        "Inworld TTS requires an apiKey. Pass { apiKey: '...' } or " +
          "set INWORLD_API_KEY in the environment.",
      );
    }
    const { apiKey: _ignored, ...rest } = opts;
    void _ignored;
    super(key, rest);
  }
}

File: apps/patter-voice-sdk/libraries/python/getpatter/tts/inworld.py
"""Inworld TTS for Patter pipeline mode."""

from __future__ import annotations

import os
from typing import ClassVar, Optional

from getpatter.providers.inworld_tts import InworldTTS as _InworldTTS

__all__ = ["TTS"]


class TTS(_InworldTTS):
    """Inworld HTTP NDJSON TTS — defaults to the TTS-2 model.

    Example::

        from getpatter.tts import inworld

        tts = inworld.TTS()                                    # reads INWORLD_API_KEY
        tts = inworld.TTS(api_key="...", voice="Olivia", language="en")
    """

    provider_key: ClassVar[str] = "inworld"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        model: str = "inworld-tts-2",
        voice: str = "Ashley",
        language: Optional[str] = None,
        audio_encoding: str = "PCM",
        sample_rate: int = 16000,
        bitrate: int = 64000,
        temperature: Optional[float] = None,
        speaking_rate: float = 1.0,
        delivery_mode: Optional[str] = None,
    ) -> None:
        key = api_key or os.environ.get("INWORLD_API_KEY")
        if not key:
            raise ValueError(
                "Inworld TTS requires an api_key. Pass api_key='...' or "
                "set INWORLD_API_KEY in the environment."
            )
        super().__init__(
            auth_token=key,
            model=model,
            voice=voice,
            language=language,
            audio_encoding=audio_encoding,
            sample_rate=sample_rate,
            bitrate=bitrate,
            temperature=temperature,
            speaking_rate=speaking_rate,
            delivery_mode=delivery_mode,
        )
